lib/bundle/brew_installer.rb

```diff
@@ -14,21 +14,35 @@ def initialize(name, options = {})
       @full_name = name
       @name = name.split("/").last
       @args = options.fetch(:args, []).map { |arg| "--#{arg}" }
+      @unlink = options.fetch(:unlink, false)
+      @stop_service = options.fetch(:stop_service, false)
       @restart_service = options.fetch(:restart_service, false)
     end
 
     def run
-      install_or_upgrade and restart_service!
+      install_change_state! && service_change_state!
     end
 
-    def install_or_upgrade
+    def unlink?
+      @unlink
+    end
+
+    def install_change_state!
       if installed?
-        upgrade!
+        if unlink?
+          unlink!
+        else
+          upgrade!
+        end
       else
         install!
       end
     end
```

Comment:
> This is very counter-intuitive.

Agreed, will fix.



> From what I'm understanding, what you really need is an unlink-if-conflict feature. This also can be applied to services. I think what you want is stop-services-if-conflict feature(this may be better to be handled inside brew services).

That relies on conflicts being declared consistently and correctly and I don't think we can rely on that particularly when e.g. you're mixing third-party taps.